app/scripts/lib/derived-keys.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Derive relier-specific encryption keys from account master keys.
+ */
+
+'use strict';
+
+define([
+  'sjcl',
+  'lib/hkdf',
+  'lib/constants'
+], function (sjcl, hkdf, Constants) {
+
+  /**
+   * Given the account master keys, the user id and the relier id, generate
+   * the matching relier-specific derived keys.
+   */
+  function deriveRelierKeys(keys, uid, rid) {
+    var relierKeys = {};
+    return generateDerivedKey(keys.kA, 'kAr', uid, rid)
+      .then(function (kAr) {
+        relierKeys.kAr = kAr;
+        return generateDerivedKey(keys.kB, 'kBr', uid, rid);
+      }).then(function (kBr) {
+        relierKeys.kBr = kBr;
+        return relierKeys;
+      });
+  }
+
+  /**
+   * Given master key material, the name of the key class in question,
+   * the user id and the relier id, generate the appropriate derived key
+   * as a JWK object.
+   */
+  function generateDerivedKey(key, kcls, uid, rid) {
+    key = sjcl.codec.hex.toBits(key);
+    var info = Constants.DERIVED_KEY_INFO_PREFIX + kcls + ':' + rid;
+    info = sjcl.codec.utf8String.toBits(info);
```

Comment:
"info" is the term used in the definition of HKDF; I'll shuffle the code a bit and rename it to something like "contextInfo" which should make things a little clearer.